node/src/tasks/bsp_volunteer_mock.rs

```diff
@@ -0,0 +1,118 @@
+                multiaddresses,
+            });
+
+        let (mut tx_watcher, tx_hash) = self
+            .storage_hub_handler
+            .blockchain
+            .send_extrinsic(call)
+            .await;
+
+        // Wait for the transaction to be included in a block.
+        let mut block_hash = None;
+        // TODO: Consider adding a timeout.
+        while let Some(tx_result) = tx_watcher.recv().await {
+            // Checking if there is an update with an error in our transaction.
+            if tx_result.starts_with("Error") {
+                error!(target: LOG_TARGET, "Error in transaction: {:?}", tx_result);
+                return Err(anyhow::anyhow!("Error in transaction: {:?}", tx_result));
+            }
+
+            // Parse the JSONRPC string, now that we know it is not an error.
+            let json: serde_json::Value = serde_json::from_str(&tx_result)
+                .expect("The result, if not an error, can only be a JSONRPC string; qed");
+
+            debug!(target: LOG_TARGET, "Transaction information: {:?}", json);
+
+            // Checking if the transaction is included in a block.
+            // TODO: Consider if we might want to wait for "finalized".
+            if let Some(in_block) = json["params"]["result"]["inBlock"].as_str() {
+                block_hash = Some(H256::from_str(in_block)?);
+                let subscription_id = json["params"]["subscription"]
+                    .as_number()
+                    .expect("Subscription should exist and be a number; qed");
+
+                // Unwatch extrinsic to release tx_watcher.
+                self.storage_hub_handler
+                    .blockchain
+                    .unwatch_extrinsic(subscription_id.to_owned())
+                    .await?;
+
+                // Breaking while loop.
+                // Even though we unwatch the transaction, and the loop should break, we still break manually
+                // in case we continue to receive updates. This should not happen, but it is a safety measure,
+                // and we already have what we need.
+                break;
+            }
+        }
+
+        // Get the extrinsic from the block, with its events.
+        let block_hash = block_hash.expect(
+            "Block hash should exist after waiting for extrinsic to be included in a block; qed",
+        );
+        let extrinsic_in_block = self
+            .storage_hub_handler
+            .blockchain
+            .get_extrinsic_from_block(block_hash, tx_hash)
+            .await?;
+
+        // Check if the extrinsic was successful. In this mocked task we know this should fail if Alice is
+        // not a registered BSP.
+        let extrinsic_successful = ActorHandle::<BlockchainService>::extrinsic_successful(extrinsic_in_block.clone())
+            .expect("Extrinsic does not contain an ExtrinsicFailed nor ExtrinsicSuccess event, which is not possible; qed");
+        if !extrinsic_successful {
```

Comment:
Regarding my previous comment, we could have access to `dispatch_err` and `dispatch_info` if we add it in the return type of `extrinsic_successful`. It would be useful.

Reply:
If you see the logs now, you'll find the error, although since it is encoded, it won't be of much use as it is. But it's there.